packages/public/server/src/module/Common/src/Filter/Slugify.php

```diff
@@ -28,34 +28,51 @@ class Slugify implements FilterInterface
 {
     /**
      * @param string $text
-     * @return bool|mixed
+     * @return bool|string
      */
-    protected static function slugify($text)
+    protected static function slugify(string $text)
     {
-        $text = trim($text, ' ');
-        $text = preg_replace('~ +~u', '-', $text);
+        $replacementTable = [
+            // remove
+            "'" => '',
+            '"' => '',
+            '`' => '',
+            '=' => '',
+            '+' => '',
+            '*' => '',
+            '&' => '',
+            '^' => '',
+            '%' => '',
+            '$' => '',
+            '#' => '',
+            '@' => '',
+            '!' => '',
+            '<' => '',
+            '>' => '',
+            '?' => '',
+            // convert to minus
+            '[' => '-',
+            ']' => '-',
+            '{' => '-',
+            '}' => '-',
+            '(' => '-',
+            ')' => '-',
+            ' ' => '-',
+            ',' => '-',
+            ';' => '-',
+            ':' => '-',
+            '/' => '-',
+            '|' => '-',
+        ];
+        $text = strtr($text, $replacementTable);
+        $text = preg_replace('/-{2,}/', '-', $text);
         $text = strtolower($text);
         if (empty($text)) {
             return false;
         }
         return $text;
     }
 
-    /**
-     * @param string $str
-     * @param string $delimiter
-     * @return mixed|string
-     */
-    protected static function toAscii($str, $delimiter = '-')
-    {
-        $clean = iconv('UTF-8', 'ASCII//TRANSLIT', $str);
-        $clean = preg_replace('/[^a-zA-Z0-9\/_|+ -]/', '', $clean);
-        $clean = strtolower(trim($clean, '-'));
-        $clean = preg_replace('/[\/_|+ -]+/', $delimiter, $clean);
-
-        return $clean;
-    }
```

Comment:
Sidenote: I want to have a dead code detector so that we can get rid of all the dead code in this repo. Thanks for deleting :tada: :smile:

Reply:
Well, similar to the stuff above: since we don't have tests in place, this is always a little tricky since PHP (and Zend especially) doesn't always use the stuff in a way that a dead code detector recognizes it (e.g. `fooAction` in a controller is mentioned as `foo` in the `module.config.php`. Similarly, except in newer code, we refer to classes via a string (sometimes even a dynamic string using `__NAMEPACE__`)...)